### ISSUE TITLE
Rule proposal: Require escape sequences to use lowercase values

### ISSUE THREAD

**Issue by otomad** [sindresorhus/eslint-plugin-unicorn#2078]
### Description

The `escape-case` and `number-literal-case` rules supported uppercase values now, however is it possible to add a **lowercase option** for it?

### Fail

```js
// unicorn/escape-case
const foo = '\xA9';
const foo = '\uD834';
const foo = '\u{1D306}';
const foo = '\cA';

// number-literal-case
const foo = 0xA9;
const foo = 0xFF;
```

### Pass

```js
// unicorn/escape-case
const foo = '\xa9';
const foo = '\ud834';
const foo = '\u{1d306}';
const foo = '\ca';

// number-literal-case
const foo = 0xa9;
const foo = 0xff;
```

### Additional Info

The config can be like this:
```json
{
    "rules": {
        "unicorn/escape-case": ["error", "lowercase"],
	"unicorn/number-literal-case": ["error", "lowercase"],
    }
}
```

**Comment by matzkoh** [comment]
`number-literal-case` seems to conflict with Prettier.
I prefer Prettier's lowercase conversion. So it would be nice if these rules had a lowercase option.

**Comment by fisker** [comment]
I'm fine to add an option.
